src/Raven.Server/Documents/Sharding/Handlers/Processors/Revisions/ShardedRevisionsHandlerProcessorForGetRevisions.cs

```diff
@@ -36,15 +37,16 @@ protected override async ValueTask GetRevisionByChangeVectorAsync(TransactionOpe
                 return;
             }
 
-            string etag = null; //TODO
-            if (NotModified(etag))
+            var revisions = ShardResultConverter.BlittableToDocumentsListConverter(res);
+            var actualEtag = ComputeHttpEtags.ComputeEtagForRevisions(revisions);
```

Comment:
I wonder if we need to work so hard... would xor on the etag headers from the requests would be easier?

Reply:
Not sure I follow.. We don't have the headers here...

Reply:
You can get it (or maybe compute it?) from the `ShardedGetRevisionsByChangeVectorsOperation` above... no?